### PR TITLE
VPC: Clean up vagrant boxes on exit

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

ref https://github.com/adoptium/infrastructure/issues/2581

### CODE REVIEW

**Comment by Haroon-Khel** [comment]
Clean up looks like it has improved.
For unstable and failed jobs, the vpc will still attempt to destroy the vm (unless the user has chosen to keep the vm)
https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/1451/OS=Ubuntu1804,label=vagrant/console is an example; failed at the test stage yet still destroyed its vm

**Comment by Haroon-Khel** [comment]
https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/OS=Debian8,label=vagrant/1452/console
Clean up occurs on abort too. 

I can confirm that the '[leftover processes](https://github.com/adoptium/infrastructure/issues/2581#issue-1249389160)' disappear when the vagrant box is cleared up

**Comment by Haroon-Khel** [comment]
For reviewers, look at the changed files with the 'hide whitespace' option for a cleaner view

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh [hunk 1]
```diff
@@ -371,26 +371,33 @@ startVMPlaybookWin()
...
-	free
-	echo === showing global status while pruning:
-	vagrant global-status --prune
-	echo === Determining VM to destroy:
-	VM_TO_DESTROY=`vagrant global-status --prune | grep $OS | awk "/${gitFork}-${gitBranch}/ { print \\$1 }"`
-	if [ ! -z "$VM_TO_DESTROY" ]; then
-	  echo === Destroying VM with id $VM_TO_DESTROY
-	  vagrant destroy -f $VM_TO_DESTROY
+	if [[ "$retainVM" == false ]]; then
+		echo "Destroying the VM"
```
Comment (sxa) [inline ansible/pbTestScripts/vagrantPlaybookCheck.sh:375]: Do we really need this echo statement given there's another more descriptive one two lines down?

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh [hunk 2]
```diff
@@ -371,26 +371,33 @@ startVMPlaybookWin()
...
 	else
-	  echo === NOT DESTROYING ANY VM as no suitable ID was found searching for $OS and ${gitFork}-${gitBranch}
+		echo "You have chosen to retain the VM. It will not be destroyed"
 	fi
-        echo === Final status:
-        vagrant global-status --prune
-        free
 }
 
+trap 'destroyVM $vagrantOS' EXIT
```
Comment (sxa) [inline ansible/pbTestScripts/vagrantPlaybookCheck.sh:399]: Isn't vagrantOS potentially a list? That's why the original call to destroyVM is `$OS` as that's the loop iterator over `$vagrantOS` later in the script.

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh (hunk 1 above)
Comment (Haroon-Khel) [reply ansible/pbTestScripts/vagrantPlaybookCheck.sh:375]: I added this as a debug line, forgot to remove it :L

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh (hunk 2 above)
Comment (Haroon-Khel) [reply ansible/pbTestScripts/vagrantPlaybookCheck.sh:399]: It is potentially, but as far as I know it is never used as a list. Our VPC job only ever kicks off one OS at a time (separates multiple OSs over single jobs). But I'll make some adjustments

**Review by Willsparker — APPROVED** [review]
Agreed with @sxa that it should be able to handle them as a list. The idea behind it was that you can do a kind-of batch job, if using it from CLI, as opposed to Jenkins. It looks like your most recent commit will work, for both 👍 
LGTM

**Pull request merged**